src/components/TrackList.test.js

```diff
@@ -0,0 +1,168 @@
+                auxPallete: "reds",
+                colorReadsByMappingQuality: false
+            }
+        },
+    }
+    const availableColors = ["greys", "ygreys", "reds", "plainColors", "lightColors"];
+    const availableTracks = [{"files": [{"name": "fileA1.vg", "type": "graph"},
+                                        {"name": "fileA2.gbwt", "type": "haplotype"}]},
+                             {"files": [{"name": "fileB1.gbwt", "type": "haplotype"},
+                                        {"name": "fileB2.gam", "type": "read"}]},
+                             {"files": [{"name": "fileC1.xg", "type": "graph"}]}];
+
+    function rerenderTrackList(rerender, newTracks, fakeOnChange, fakeOnDelete) {
+        rerender(
+            <TrackList
+                tracks={newTracks}
+                availableTracks={availableTracks}
+                availableColors={availableColors}
+                onChange={fakeOnChange}
+                onDelete={fakeOnDelete}
+            />
+        );
+    }
+
+
+    it('should render without errors', async () => {
+        const fakeOnChange = jest.fn();
+        const fakeOnDelete = jest.fn();
+        const { queryByTestId } = render(
+            <TrackList
+                tracks={tracks}
+                availableTracks={availableTracks}
+                availableColors={availableColors}
+                onChange={fakeOnChange}
+                onDelete={fakeOnDelete}
+            />
+        );
+
+        expect(queryByTestId("file-type-select-component1")).toBeTruthy();
+
+        expect(queryByTestId("file-type-select-component2")).toBeTruthy();
+
+        expect(queryByTestId("file-select-component1")).toBeTruthy();
+
+        expect(queryByTestId("settings-button-component1")).toBeTruthy();
+
+        expect(queryByTestId("delete-button-component1")).toBeTruthy();
+        
+    });
+
+    it('should call onChange when a track is changed', async () => {
+        const fakeOnChange = jest.fn();
+        const fakeOnDelete = jest.fn();
+        const { queryByTestId, getByText, rerender } = render(
+            <TrackList
+                tracks={tracks}
+                availableTracks={availableTracks}
+                availableColors={availableColors}
+                onChange={fakeOnChange}
+                onDelete={fakeOnDelete}
+            />
+        );
+
+        // select file type
+        const fileTypeSelectComponent = queryByTestId("file-type-select-component1");
+        fireEvent.keyDown(fileTypeSelectComponent.firstChild, {key: "ArrowDown"});
+        await waitFor(() => getByText("haplotype"));
+        fireEvent.click(getByText("haplotype"));
+
+        // should wait for file to be selected
+        expect(fakeOnChange).toHaveBeenCalledTimes(0);
+
+        // simulate dropdown effect
+        let newTracks = {...tracks};
+        newTracks[1].trackType = "haplotype";
+
+        rerenderTrackList(rerender, newTracks, fakeOnChange, fakeOnDelete);
```

Comment:
Why is this step needed? The `TrackListItem` component is supposed to take care of holding its own state for how haplotypes has been selected but a file hasn't yet been picked, right? In the real page the track type in the list won't have changes yet, so why change it in the test?